language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/docs/NodeBasedUpdateDocumentationCodeAction.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ballerinalang.langserver.codeaction.providers.docs;
+
+import io.ballerina.compiler.api.SemanticModel;
+import io.ballerina.compiler.api.symbols.Documentable;
+import io.ballerina.compiler.api.symbols.Documentation;
+import io.ballerina.compiler.api.symbols.Symbol;
+import io.ballerina.compiler.syntax.tree.NonTerminalNode;
+import io.ballerina.compiler.syntax.tree.SyntaxTree;
+import org.ballerinalang.annotation.JavaSPIService;
+import org.ballerinalang.langserver.codeaction.providers.AbstractCodeActionProvider;
+import org.ballerinalang.langserver.command.docs.DocAttachmentInfo;
+import org.ballerinalang.langserver.command.executors.UpdateDocumentationExecutor;
+import org.ballerinalang.langserver.common.constants.CommandConstants;
+import org.ballerinalang.langserver.common.utils.CommonUtil;
+import org.ballerinalang.langserver.commons.CodeActionContext;
+import org.ballerinalang.langserver.commons.codeaction.CodeActionNodeType;
+import org.ballerinalang.langserver.commons.codeaction.spi.NodeBasedPositionDetails;
+import org.ballerinalang.langserver.commons.command.CommandArgument;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.Command;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static org.ballerinalang.langserver.command.docs.DocumentationGenerator.getDocumentableSymbol;
+import static org.ballerinalang.langserver.command.docs.DocumentationGenerator.getDocumentationEditForNode;
+import static org.ballerinalang.langserver.command.docs.DocumentationGenerator.hasDocs;
+
+/**
+ * Code Action for updating single documentation.
+ *
+ * @since 2.0.0
+ */
+@JavaSPIService("org.ballerinalang.langserver.commons.codeaction.spi.LSCodeActionProvider")
+public class NodeBasedUpdateDocumentationCodeAction extends AbstractCodeActionProvider {
```

Comment:
@IMS94 as per the design, even if we can use the both the `getNodeBasedCodeActions` and `getDiagnosticBasedCodeActions` in the same class, we do not encourage to do so, in order to maintain the single responsibility practices. Also, We have to keep the diagnostic based code action since we can support update documentation through the diagnostics shown on the parameters. 
Hence let's keep both for the moment and do an evaluation before removing. WDYT?

Reply:
@nadeeshaan Agreed. There can be scenarios which are not just mismatches, but best practices reported via diagnostics. In such cases, we may need the diagnostics based one. Otherwise, almost all the cases will be covered by the node based one.